specs/Deployment/DeploymentAPI.md

```diff
@@ -0,0 +1,136 @@
+declared the WAS frameworks as a dependency. In this scenario it is likely the main package would
+be missing because they cannot declare a dependency on the main package and it will not be
+deployed by the Store. For MSIX packages with permissions, this dependency could be immediately
+resolved online or offline if the package size is small and the framework carries the dependency.
+
+A class of MSIX packages that are Full Trust or have package management capabilities, such as
+Centennial MSIX packages will have the permissions to immediately resolve the dependency. These
+packages are the target for this version of the API. These packages will have the capability
+to deploy packages and satisfy the dependency. Since the Main Package for WAS is expected to be
+very small, this API has a streamlined solution for this scenario, which is to carry the WAS
+main packages inside the WAS framework, and leverage the full trust capability of the package
+to deploy the main package. A later release will address how to resolve the missing packages
+for UWPs when they become supported.
+
+
+# Description
+
+When a Windows App SDK (WAS) consuming Centennial package is deployed from the Store, it is
+expected to only have the App package and frameworks, and the WAS main package may not be present,
+or may not be at the desired version. This API enables a developer to detect when the required WAS
+main and/or framework packages are not present and take corrective action.
+
+Since this API is part of the WAS framework, the framework itself must be present.
+For MSIX packages, the framework must be listed as a dependency in the application manifest and
+will not be deployable without a min version being present. 
+
+Developers must not assume that the WAS main package is present, or that the version
+present is the minimum version required to support the feature set they expect. Instead, this API
+should be used to validate the WAS is the correct version and all required packages are present.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of the Windows App SDK I require?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Since the main packages are small and the target developer
+scenario will have permissions to deploy them, we will minimize developer and user pain by
+including the main package MSIX packages within the framework, and this API will allow for the
+deployment of that package if needed.
+
+For this release, packages without permission to deploy are not supported, and future versions
+will have solutions for such packages.
+
+# Examples
+
+## GetStatus
+
+This is the means by which an app or program can call to verify the runtime packages is present to
```

Comment:
packages <del>is</del> are present

Reply:
why have GetStatus at all if Initialize can check status _and_ then resolve dependencies if any are missing?

Reply:
GetStatus is essentially a exists check that will tell you if there is any work to do. Since the Initialize() method may do package installs, and package installs can take significant amounts of time, the GetStatus method is useful because a developer may wish to defer that potential performance-impacting work to a time that is not performance sensitive. Additionally, the developer may wish to gate functionality on availability of the WinAppSDK if they are using it for a limited set. This allows a low-impact way to test for that without attempting to do package deployment.